zamba/models/config.py

```diff
@@ -481,78 +481,105 @@ def validate_filepaths_and_labels(cls, values):
 
     @root_validator(skip_on_failure=True)
     def preprocess_labels(cls, values):
+        """One hot encode, add splits, and check for binary case.
+
+        Replaces values['labels'] with modified DataFrame.
+
+        Args:
+            values: dictionary containing labels and other config info
+        """
         logger.info("Preprocessing labels into one hot encoded labels with one row per video.")
+        labels = values["labels"]
+
         # one hot encode collapse to one row per video
+        # TODO: replace this with a call to one_hot_encode?
```

Comment:
we could, but that [function](https://github.com/drivendataorg/pjmf-zamba/blob/main/src/data/metadata.py#L219-L247) (which lives in a separate repo so would need to be copied over) does some merging back in, assuming that there are non-label metadata columns provided. that will be the case if the user provides site or split, but if the user just provides filepath and label then that code as written will fail. i'm inclined to lave the get_dummies code as is

aside: i wonder what happens if a user provides conflicting site or split information, e.g.
```
filepath,label,site
vid_1.mp4,duiker,site_a
vid_1.mp4,bird,site_b
```
i don't think we check for that and just take the "max" for that video in this groupby (which yields one site per video). this seems like a fine assumption, though we could check this earlier and raise and error instead